src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -784,6 +812,266 @@ private void PopulateEncLogTableParameters()
+            var rowId = 1;
+            // The index into emittedAttributeParents, that is the thing we're currently looking for
+            var emittedIndex = 0;
+            // The index in encLogRows we're going to update
+            var encLogRowId = 0;
+
+            // Create a list to store the parents we skip, that come from subsequent generations
+            var skippedParents = new List<EntityHandle>();
+
+            // We iterate through the original metadata, and the emitted attributes at the same time, since they're sorted the same.
+            // metadataReader.CustomAttributes isn't indexable so we're enumerating manually in order to have control of things.
+            using var enumerator = metadataReader.CustomAttributes.GetEnumerator();
+            var hasData = enumerator.MoveNext();
+
+            // We want to manually control where we are in the list of custom attributes so we only MoveNext at the end
+            while (hasData)
+            {
+                // Data from the existing CA table
+                var customAttributeHandle = enumerator.Current;
+                var parent = metadataReader.GetCustomAttribute(customAttributeHandle).Parent;
+                var parentCodedIndex = CodedIndex.HasCustomAttribute(parent);
+
+                // Data from the CA table delta we are emitting
+                var emittedParent = emittedAttributeParents[emittedIndex];
+                var emittedCodedIndex = CodedIndex.HasCustomAttribute(emittedParent);
+
+                if (parentCodedIndex > emittedCodedIndex)
+                {
+                    // If the existing data is ahead of the emitted data then it means the current emitted attribute
+                    // must be for something that wasn't in the original compilation so we'll deal with it later.
+                    skippedParents.Add(emittedParent);
+
+                    // Move our log row index on enough to skip over the remaining attributes for this parent, which will
+                    // be filled in later when we process the deltas
+                    encLogRowId += _customAttributeParentCounts[emittedParent];
+
+                    // We're skipping the emitted attribute, so move that index on, but we want to re-check the
+                    // existing attribute in case it matches our next emitted attribute
+                    emittedIndex++;
+                }
+                else if (parent == emittedParent)
+                {
+                    // If we found the attribute we want then add an EncLog table record for it in the right position
+                    // If the right position doesn't exist, then that means we found a parent for which an attribute has
+                    // been deleted
+                    if (encLogRows.Count <= encLogRowId)
+                    {
+                        // Expand our enc log table to accommodate
+                        encLogRows.Add(0);
```

Comment:
>encLogRows.Add(0); [](http://example.com/codeflow?start=24&length=18)

nit: rather then adding 0 and then immediately overwrite it with RowId it would be clearer to juse add rowId right away and move the next line to `else`